### PR TITLE
[TR] PIM-4437: re-work import cookbooks

### PR DESCRIPTION
**Pull request by nidup**

### CODE REVIEW

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
+How to Clean a CSV File During a Products Import
+================================================
```
Comment (rybus) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: Product ?

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+An item step is configured with 3 elements, a reader, a processor and a writer.
+
+Here, we'll use a custom processor service, named ``acme_csvcleanerconnector.processor.denormalization.product.flat`` but we'll continue to use default reader and writer.
+
+.. important::
+
+    We strongly advise to always try to re-use most of existing pieces, it ensures that all business rules and validation will be applied.
+
+Configure our Processor
```
Comment (rybus) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: the

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+-----------------------
+
+In fact, we're using the default processor class but we create a new service to change the injected array converter (replace ``pim_connector.array_converter.flat.product`` by ``acme_csvcleanerconnector.array_converter.flat.product``), all the other services remain the same.
+
+.. literalinclude:: ../../src/Acme/Bundle/CsvCleanerConnectorBundle/Resources/config/processors.yml
+   :language: yaml
+   :linenos:
+
+Create our ArrayConverter
```
Comment (rybus) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: the

File: cookbook/import_export/product-import-from-xml-file.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
+How to Import Products from a XML File
+======================================
+
+The foundations of connector creation have been covered in the previous chapter (cf :doc:`/cookbook/import_export/create-connector`). With the following hands-on practice, we will create our own specific connector.
```
Comment (rybus) [inline cookbook/import_export/product-import-from-xml-file.rst]: The foundation or foundations but not both

File: cookbook/import_export/product-import-from-xml-file.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+The default step is ``Akeneo\Bundle\BatchBundle\Step\ItemStep``.
+
+An item step is configured with 3 elements, a reader, a processor and a writer.
+
+Here, we'll use a custom reader ``acme_xml_connector.reader.file.xml_product`` but we'll continue to use default processor and writer.
+
+.. important::
+
+    We strongly advise to always try to re-use most of existing pieces, especially processor and writer, it ensures that all business rules and validation will be properly applied.
```
Comment (rybus) [inline cookbook/import_export/product-import-from-xml-file.rst]: most of the

File: cookbook/import_export/product-import-from-xml-file.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+Create our Reader
+-----------------
+
+As we don't have existing reader which allows to read this kind of file, we'll write a new reader.
```
Comment (rybus) [inline cookbook/import_export/product-import-from-xml-file.rst]: an existing

File: cookbook/import_export/product-import-from-xml-file.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+The purpose of the reader is to return each item as an array, in the case of XML file, we can have more work to define what is the item.
+
+.. literalinclude:: ../../src/Acme/Bundle/XmlConnectorBundle/Reader/File/XmlProductReader.php
+   :language: php
+   :linenos:
+
+Our element reads the file and iterates to return products line by line.
+
+This element must be configured with the path of the xml file (an example file is provided in ``XmlConnectorBundle\Resources\fixtures\products.xml``).
```
Comment (rybus) [inline cookbook/import_export/product-import-from-xml-file.rst]: XML was written in caps on the previous sentence

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/batch_jobs.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+        dummy_job:
+            title: acme_dummy_connector.jobs.dummy_job.title
+            type:  export
+            steps:
+                dummy_step:
+                    title:         acme_dummy_connector.jobs.dummy_job.export.title
+                    services:
+                        reader:    pim_connector.reader.dummy_item
+                        processor: pim_connector.processor.dummy_item
+                        writer:    pim_connector.writer.dummy_item
```
Comment (willy-ahva) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/batch_jobs.yml]: blank line

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 9]
```diff
@@ -0,0 +1,112 @@
+How to Clean a CSV File During a Products Import
+================================================
+
+The foundations of connector creation have been covered in the previous chapter (cf :doc:`/cookbook/import_export/create-connector`). With the following hands-on practice, we will create our own specific connector.
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: Foundations

Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: a specific connector

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+To stay focused on the main concepts, we will implement the simplest connector possible by avoiding to use too many existing elements.
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: main concepts

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+Our use case is to clean the following CSV file when importing products:
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: The use case

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+Here, we want remove the prefix ``uselesspart-`` in the sku before to run a classic import.
+
+We assume that we're using a standard edition with the ``icecat_demo_dev`` data set, ``sku`` and ``name`` already exist as real attributes of the PIM.
+
+.. note::
+
+    The code inside this cookbook entry is available in the src directory, you can clone pim-docs (https://github.com/akeneo/pim-docs) and use a symlink to make the Acme bundle available in the `src/`.
+
+Create our Connector
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: "the Connector"

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+    :linenos:
+
+    public function registerBundles()
+    {
+        // ...
+            new Acme\Bundle\CsvCleanerConnectorBundle\AcmeCsvCleanerConnectorBundle(),
+        // ...
+    }
+
+Configure our Job
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: the job

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+   :linenos:
+   :lines: 1-13
+
+Here we create an import job which contains a single step: `import`.
+
+The default step is ``Akeneo\Bundle\BatchBundle\Step\ItemStep``.
+
+An item step is configured with 3 elements, a reader, a processor and a writer.
+
+Here, we'll use a custom processor service, named ``acme_csvcleanerconnector.processor.denormalization.product.flat`` but we'll continue to use default reader and writer.
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: service named ``acme_csvcleanerconnector.processor.denormalization.product.flat``,(comma here) but we

File: cookbook/import_export/clean-csv-file-during-product-import.rst [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+.. important::
+
+    We strongly advise to always try to re-use most of existing pieces, it ensures that all business rules and validation will be applied.
+
+Configure our Processor
+-----------------------
+
+In fact, we're using the default processor class but we create a new service to change the injected array converter (replace ``pim_connector.array_converter.flat.product`` by ``acme_csvcleanerconnector.array_converter.flat.product``), all the other services remain the same.
```
Comment (willy-ahva) [inline cookbook/import_export/clean-csv-file-during-product-import.rst]: all other services

File: cookbook/import_export/product-import-from-xml-file.rst (hunk 4 above)
Comment (nidup) [inline cookbook/import_export/product-import-from-xml-file.rst]: sorry, I don't understand

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/batch_jobs.yml (hunk 8 above)
Comment (nidup) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/batch_jobs.yml]: fixed

**Pull request merged**